File: src/main/java/co/gatedaccess/web/data/repo/CommunityRepo.java
package co.gatedaccess.web.data.repo;

import co.gatedaccess.web.data.model.Community;
import org.springframework.data.mongodb.repository.MongoRepository;

public interface CommunityRepo extends MongoRepository<Community, String> {

    Community findCommunityById(String id);

    Community findCommunityBySuperAdminId(String id);
}

File: src/main/kotlin/co/gatedaccess/web/data/repo/CommunityRepo.kt
package co.gatedaccess.web.data.repo

import co.gatedaccess.web.data.model.Community
import org.springframework.data.mongodb.repository.MongoRepository

interface CommunityRepo : MongoRepository<Community?, String?> {
    fun findCommunityById(id: String?): Community?

    fun findCommunityBySuperAdminId(id: String?): Community?
    fun findCommunityByIdAndSuperAdminId(id: String, superAdminId: String): Community?
    fun findByAdminIdsContains(adminId: String): Community?
    fun  existsByAdminIdsContains(adminId: String): Boolean
}
